v-next/hardhat/src/internal/cli/helpers/utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { ParameterType } from "@nomicfoundation/hardhat-core/config";
+import type { Task } from "@nomicfoundation/hardhat-core/types/tasks";
+
+export const GLOBAL_OPTIONS = [
+  {
+    name: "--config",
+    description: "A Hardhat config file.",
+  },
+  {
+    name: "--help",
+    description: "Shows this message, or a task's help if its name is provided",
+  },
+  {
+    name: "--show-stack-traces",
+    description: "Show stack traces (always enabled on CI servers).",
+  },
+  {
+    name: "--version",
+    description: "Shows hardhat's version.",
+  },
+];
+
+export const GLOBAL_NAME_PADDING = 6;
+
+export function parseTasks(taskMap: Map<string, Task>): {
+  tasks: Array<{ name: string; description: string }>;
+  subtasks: Array<{ name: string; description: string }>;
+} {
+  const tasks = [];
+  const subtasks = [];
+
+  for (const [taskName, task] of taskMap) {
+    subtasks.push(...parseSubtasks(task));
+
+    if (task.isEmpty) {
+      continue;
+    }
+
+    tasks.push({ name: taskName, description: task.description });
+  }
+
+  return { tasks, subtasks };
+}
+
+export function parseSubtasks(
+  task: Task,
+): Array<{ name: string; description: string }> {
+  const subtasks = [];
+
+  for (const [, subtask] of task.subtasks) {
+    subtasks.push({
+      name: subtask.id.join(" "),
+      description: subtask.description,
+    });
+  }
+
+  return subtasks;
+}
+
+export function parseOptions(task: Task): {
+  options: Array<{ name: string; description: string; type: ParameterType }>;
+  positionalArguments: Array<{ name: string; description: string }>;
+} {
+  const options = [];
+  const positionalArguments = [];
+
+  for (const [optionName, option] of task.namedParameters) {
+    options.push({
+      name: formatOptionName(optionName),
+      description: option.description,
+      type: option.parameterType,
+    });
+  }
+
+  for (const argument of task.positionalParameters) {
+    positionalArguments.push({
+      name: argument.name,
+      description: argument.description,
+    });
+  }
+
+  return { options, positionalArguments };
+}
+
+export function formatOptionName(str: string): string {
+  return `--${str
+    .split("")
+    .map((letter, idx) => {
+      return letter.toUpperCase() === letter
+        ? `${idx !== 0 ? "-" : ""}${letter.toLowerCase()}`
+        : letter;
+    })
```

Comment:
@schaable do we have a utility for this?

Reply:
we have the opposite (`kebabToCamel`). We should add this.

Reply:
Added an issue to track it https://github.com/NomicFoundation/hardhat/issues/5465